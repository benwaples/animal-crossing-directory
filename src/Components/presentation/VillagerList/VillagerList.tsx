import React from 'react'
// import styles from './VillagerList.css'
import { Villager } from '../Villager/Villager'

type Props = {
  villagers: {
    _id: string,
    name: string;
    imageUrl: string;
  }[];
}

export const VillagerList = ({ villagers }: Props) => {
  console.log(villagers)
  const villagerElements = villagers.map(villager => <Villager key={villager.name} { ...villager }/>)

  return (
    <div>
      {villagerElements}
    </div>
  )
}
