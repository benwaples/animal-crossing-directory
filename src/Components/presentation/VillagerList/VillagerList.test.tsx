import React from 'react'
import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import { VillagerList } from './VillagerList'

describe('VillagerList component snapshot', () => {
  it('should render a VillagerList Component', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <VillagerList 
        villagers={[
          {
          _id: 'random String',
          name: 'random String',
          imageUrl: 'random String',
        },
          {
          _id: 'random',
          name: 'random',
          imageUrl: 'random',
        },
          {
          _id: 'String',
          name: 'String',
          imageUrl: 'String',
        },
      ]}
        />
      </MemoryRouter>
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
