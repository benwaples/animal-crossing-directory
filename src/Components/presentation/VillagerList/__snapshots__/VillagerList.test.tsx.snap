// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VillagerList component snapshot should render a VillagerList Component 1`] = `
<DocumentFragment>
  <div>
    <div>
      <h1>
        random String
      </h1>
      <img
        alt="random String"
        src="random String"
      />
      <a
        href="/villagers/random String"
      >
        Learn More
      </a>
    </div>
    <div>
      <h1>
        random
      </h1>
      <img
        alt="random"
        src="random"
      />
      <a
        href="/villagers/random"
      >
        Learn More
      </a>
    </div>
    <div>
      <h1>
        String
      </h1>
      <img
        alt="String"
        src="String"
      />
      <a
        href="/villagers/String"
      >
        Learn More
      </a>
    </div>
  </div>
</DocumentFragment>
`;
